perf/src/sigverify.rs

```diff
@@ -41,9 +41,29 @@ const TRACER_KEY: Pubkey = Pubkey::new_from_array(TRACER_KEY_BYTES);
 const TRACER_KEY_OFFSET_IN_TRANSACTION: usize = 69;
 
 lazy_static! {
-    static ref PAR_THREAD_POOL: ThreadPool = rayon::ThreadPoolBuilder::new()
-        .num_threads(get_thread_count())
-        .thread_name(|ix| format!("sigverify_{}", ix))
+    static ref PAR_THREAD_POOL_XSMALL: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(2)
+        .thread_name(|ix| format!("sigverify_xs_{}", ix))
+        .build()
+        .unwrap();
+    static ref PAR_THREAD_POOL_SMALL: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(4)
+        .thread_name(|ix| format!("sigverify_s_{}", ix))
+        .build()
+        .unwrap();
+    static ref PAR_THREAD_POOL_MEDIUM: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(8)
+        .thread_name(|ix| format!("sigverify_m_{}", ix))
+        .build()
+        .unwrap();
+    static ref PAR_THREAD_POOL_LARGE: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(16)
+        .thread_name(|ix| format!("sigverify_l_{}", ix))
+        .build()
+        .unwrap();
+    static ref PAR_THREAD_POOL_XLARGE: ThreadPool = rayon::ThreadPoolBuilder::new()
+        .num_threads(std::cmp::max(16, get_thread_count()))
+        .thread_name(|ix| format!("sigverify_xl_{}", ix))
```

Comment:
Why not [`.with_min_len`](https://docs.rs/rayon/latest/rayon/iter/trait.IndexedParallelIterator.html#method.with_min_len) instead of this?

Also might be worth checking with rayon developers and/or email list to see if they have alternative suggestions. I am sure this is not a problem specific to solana.

Reply:
Collected some data sweeping across different pack counts using a few different solutions:
<img width="800" alt="image" src="https://user-images.githubusercontent.com/44715351/172762286-a99ac2f0-1a38-442e-abeb-a85a6a8f90be.png">
This was collected on my Macbook w/ 10 cores (i.e. max of 5 threads for verify).
- 'Vanilla' is unmodified master.
- 'Limit16' applies minimum packet length of 16 on a per batch basis (not across batches).
- 'Flatten, Collect, Limit16' attempts to apply minimum packet length of 16 across batches like so:
'''
    PAR_THREAD_POOL.install(|| {
        batches.into_par_iter()
            .flatten()
            .collect::<Vec<&mut Packet>>()
            .into_par_iter()
            .with_min_len(16)
            .for_each(|p| verify_packet(p, reject_non_vote));
    });
'''

Reply:
I am guessing 16 is still very small task size to overcome overhead of parallelization.
I would suggest to try `with_min_len` equal to 32, 64, 128, ... and look for an optimal in this way.

Reply:
Each packet takes ~500us to verify, so it amortizes quicker than you might think

This design tries to keep latency constant (up to a certain point) while varying CPU commitment. 10ms felt like an okay number to shoot for, but I don't have a great reference for what is acceptable here. Any thoughts?

Reply:
Deleted prior comment because it was wrong. Packet verification is taking ~80us, which is in the same neighborhood as rayon overhead. I'll collect some data with different min lengths